packages/story-editor/src/components/floatingMenu/menu.js

```diff
@@ -42,9 +42,19 @@ const FloatingMenu = memo(
   ) {
     useLayoutEffect(() => {
       const node = ref.current;
-      const bounds = node.getBoundingClientRect();
-      node.style.setProperty('--width', `${bounds.width.toFixed(2)}px`);
-      node.style.setProperty('--height', `${bounds.height.toFixed(2)}px`);
+      const updateSize = () => {
+        node.style.width = '';
+        const bounds = node.getBoundingClientRect();
+        node.style.setProperty('--width', `${bounds.width.toFixed(2)}px`);
+        node.style.setProperty('--height', `${bounds.height.toFixed(2)}px`);
+        node.style.width = 'var(--width)';
+      };
+      updateSize();
+      // If the menu children list changes, update the size again
+      const observer = new MutationObserver(updateSize);
+      const menu = node.querySelector('[role=menu]');
+      observer.observe(menu, { childList: true });
+      return () => observer.disconnect();
```

Comment:
Note that this was necessary, because changing the border-radius lock toggle in the design panel causes the design menu to change size, but because this previously only recalculated as the selection changed, this change in size was not incorporated. Now this observer will recalculate the size whenever a control is added to or removed from the menu.

However, if we in the future have controls that change size based on external input, we need to re-evaluate this (as this only observes whether nodes are added to or removed from the menu node).